python/tests/test_eval.py

```diff
@@ -172,3 +174,138 @@ async def test_run_test_suite_with_inputs(data: st.DataObject):
             ).all()
         case Err(e):
             assert False, f"expected Ok, got Err({e})"
+
+
+def _extract_id(
+    metric_fn: SampleEvaluationFunction[T_OutputDatum], metric_input: T_OutputDatum
+) -> str:
+    return metric_fn(metric_input).interpretation.id
+
+
+def _extract_id_pair_for_input_pair(
+    metric: SampleEvaluationFunction[str], input_pair: tuple[str, str]
+) -> tuple[str, str]:
+    return (
+        _extract_id(metric_fn=metric, metric_input=input_pair[0]),
+        _extract_id(metric_fn=metric, metric_input=input_pair[1]),
+    )
+
+
+def _extract_id_pair_for_metric_pair(
+    metric_pair: tuple[SampleEvaluationFunction[str], SampleEvaluationFunction[str]],
+    metric_input: str,
+) -> tuple[str, str]:
+    return (
+        _extract_id(metric_fn=metric_pair[0], metric_input=metric_input),
+        _extract_id(metric_fn=metric_pair[1], metric_input=metric_input),
+    )
+
+
+@st.composite
+def _sample_string_pairs(draw: st.DrawFn) -> tuple[str, str]:
+    return draw(
```

Comment:
what does draw do?

Reply:
It samples randomly from the data type you specified. Docs: https://hypothesis.readthedocs.io/en/latest/data.html#composite-strategies